### DISCUSSION TITLE
App Router Sometimes Returns RSC Data Instead of HTML

### DISCUSSION THREAD

**Discussion by danmartens** [vercel/next.js#50178] [App Router]
I've deployed a site using the app router as a DigitalOcean "App" (using the default [Node.js buildpack](https://docs.digitalocean.com/products/app-platform/reference/buildpacks/nodejs/)). Sometimes when I navigate to a page, I get the RSC data (i.e. the format used to stream changes to the client, not sure if there's an official name) instead of the HTML. Sometimes it doesn't happen until I refresh a page. It doesn't happen during local development and it also doesn't happen if I build and then run the site locally.

Has anyone else experienced this? I'm not sure what could be different about the DigitalOcean environment that would cause this. I haven't noticed a pattern to which pages present this issue and which don't. It seems to even happen on pages that the `next build` output says are fully static...

**Comment by icyJoseph** [comment]
I think this comment, and the rest afterwards are a good starting point: https://github.com/vercel/next.js/issues/48569#issuecomment-1514982815

**Reply by danmartens** [reply]
Thank you!

**Reply by icyJoseph** [reply]
Please post back if you manage to fix it.

**Reply by danmartens** [reply]
It's a small project that needed to launch this week, so we ended up switching back to the page router for now. It looks like this is the issue to follow: https://github.com/vercel/next.js/pull/49140
